Task 1 - authentication, created project template

KonradCiepluch opened the pull request:
Created authentication subpages and user page.

Adamchello commented on components/Atoms/Button/Button.tsx:
```diff
@@ -0,0 +1,22 @@
+}
+
+const Button = ({ type = 'submit', isLoading, className, label, disabled, iconClass, handleClick }: IProps) => (
+  <button type={type} onClick={handleClick} className={className} disabled={disabled}>
```
Jak masz przycisk, to ma on jakieś "podstawowe style", więc w tym przypadku warto zrobić plik scss dla tego buttona i nadać mu klasę np. styled__button i mieć te style jako podstawka, a className który przychodzi w propsach jako dodatkowa klasa. Czyli będzie to wyglądało w tym stylu className={`styled__button ${className}`}. Jeszcze kilka słów odnośnie stylów - jak mamy tekst oraz ikonkę w przyciskach (socialMediaButton) to widać, że nie są wycentrowane. Standardowym rozwiązaniem jest dodanie jakiegoś flex'a oraz wycentrowanie treści w pionie i poziomie i dodanie jakiegoś gap'a. Dodatkowym atutem przycisków byłaby jakaś mikrointerakcja - aktualnie masz zmianę koloru na hoverze, ale możesz dodać, że przycisk się skaluje na hoverze oraz na akcji active przy pomocy np. biblioteki framer-motion żeby urozmaicić aplikację

Adamchello commented on components/Atoms/FormField/FormField.tsx:
```diff
@@ -0,0 +1,35 @@
+import Link from 'next/link';
+import { useFormContext } from 'react-hook-form';
+
+import styles from 'components/Organisms/PageForm/PageForm.module.scss';
```
Tutaj masz w pewnym sensie antywzorzec. Raczej nie importuje się do atomów styli z organizmów - to atomy się pierwsze tworzy, a dopiero potem organizmy. Rozwiązaniem problemu będzie wsadzenie do tego folderu pliku ze stylami i tam przeklejenie styli z organizmu

Adamchello commented on components/Atoms/FormField/FormField.tsx:
```diff
@@ -0,0 +1,35 @@
+  if (name !== 'remember')
+    return (
+      <div className={styles.form__field}>
+        <input type={type} placeholder={placeholder} {...register(name)} className={styles.form__input} />
```
Mam jeden mały problem do inputów - podczas gdy pole jest aktywne to się zaokrąglają kąty. Raczej się przyjmuje konwencje , że albo od początku masz jakiś border-radius, albo go w ogóle nie masz,

Adamchello commented on components/Molecules/SocialMediaAuth/SocialMediaAuth.tsx:
```diff
@@ -0,0 +1,61 @@
+  const buttons = providers.map(({ provider, label, iconClass }) => (
+    <div key={label} className={styles['form__media-field']}>
+      <span>lub</span>
+      <Button
```
Odnośnie przycisku to zmieniłbym trochę kolor tła bo może być nie widoczny dla użytkowników bo zmywa się z białym. Ta stronka przepisy.pl fajnie rozwiązało ten problem dawając przyciskowi kolor socialMedia i myślę, że to nie jest taki zły pomysł żeby tekst oraz ikonka były białe a przycisk był koloru wybranego socialMedia (niebieski, czerwony, czarny)

Adamchello reviewed (approved):
Logika wygląda spoko, tylko kilka mam kilka problemów ze stylami które możesz zmienić i już zacząć dodawać jakieś mikrointerakcje do strony żeby ją trochę urozmaicić (np. z pomocą framer-motion)

KonradCiepluch commented:
Fixes after CR, added animations provided by framer-motion library, hover/active for buttons, and slides for auth subpages, moved auth subpages to Auth directory in Templates. What do you think about auth subpages animations?

Adamchello reviewed (approved):
It looks better now. About the animation of subpages, you can stay with it. I have nothing to complain about on this thread
